Updates on test timeouts and flow when throughput is low

savex opened the pull request:
## Backports Required

- [ ] none - not a bug fix
- [ ] none - this is a backport
- [x] none - issue does not exist in previous branches
- [ ] none - papercut/not impactful enough to backport
- [ ] v23.2.x
- [ ] v23.1.x
- [ ] v22.3.x

## Release Notes

* none

andrewhsu commented on tests/rptest/redpanda_cloud_tests/high_throughput_test.py:
```diff
@@ -489,7 +489,7 @@ def test_max_connections(self):
         _target_total = int(self._advertised_max_client_count)
         _target_per_node = int(_target_total // len(self.cluster.nodes))
-        _target_total = int(_target_total * 0.99)
+        _target_total = int(_target_total * 0.9)
```
dropping to 90% seems aggressive to me, but i don't have a clear understanding of the metadata overhead. i think it would be more conservative to drop to 95% because i have a hard time believing we have that much metadata overhead.

also, all the comment about metadata needs to be update

savex replied:
Yes, this is not good. I've updated the test to actually push through all of the connections and detect how much concurrent ones can be achieved. See next commit

andrewhsu reviewed (commented):
git commit 87bc95f044d551c0377e95fecdc7b42eb614b666 lgtm, but the changes to `test_max_connections()` regarding `_target_total` i'm not sure about. i've seen arm tier 5 test pass with updated install pack `23.2.20231110012553`

for tier 5, the broker is set to `kafka_connections_max: '30100'` and advertised max client count is `30000` so this ends up with the test's `_target_total` to `29700` taking into account 1% metadata overhead.  @ballard26 should we account for more than 1% buffer for metadata connections so `ProducerSwarm` does not overload the target cluster?

vbotbuildovich commented:
<!--NEW FAILURES-->
new failures in https://buildkite.com/redpanda/redpanda/builds/40896#018bbbf5-214d-4ad5-9728-b542ddce15b6:
```
"rptest.tests.partition_movement_test.PartitionMovementTest.test_moving_not_fully_initialized_partition.num_to_upgrade=0"
```
<!--d195d8de-70c0-4122-b68e-d00652361569-->

savex commented:
Updated waiting and removed cutout of connection counting. Now connections are counted until last message is sent.
Here is test output for Arm-tier-5. Do not mind % calculation error, it is fixed.
```
ubuntu@ip-172-31-0-110:~/tests$  cd /home/ubuntu/tests ; /usr/bin/env /bin/python3 /home/ubuntu/.vscode-server/extensions/ms-python.python-2023.21.13141007/pythonFiles/lib/python/debugpy/adapter/../../debugpy/launcher 52519 -- -m ducktape --compress --cluster=ducktape.cluster.json.JsonCluster --cluster-file=cluster.json --globals=globals.json --max-parallel=1 --repeat=1 --test-runner-timeout=86400000 rptest/redpanda_cloud_tests/high_throughput_test.py::HighThroughputTest.test_max_connections 
[INFO:2023-11-11 02:56:18,135]: starting test run with session id 2023-11-11--012...
[INFO:2023-11-11 02:56:18,135]: running 1 tests...
[INFO:2023-11-11 02:56:18,135]: Triggering test 1 of 1...
[INFO:2023-11-11 02:56:19,235]: RunnerClient: Loading test {'directory': '/home/ubuntu/redpanda/tests/rptest/redpanda_cloud_tests', 'file_name': 'high_throughput_test.py', 'cls_name': 'HighThroughputTest', 'method_name': 'test_max_connections', 'injected_args': None}
[INFO:2023-11-11 02:56:19,240]: RunnerClient: rptest.redpanda_cloud_tests.high_throughput_test.HighThroughputTest.test_max_connections: on run 1/1
[INFO:2023-11-11 02:56:22,122]: RunnerClient: rptest.redpanda_cloud_tests.high_throughput_test.HighThroughputTest.test_max_connections: Setting up...
[WARNING - 2023-11-11 02:56:22,122 - redpanda_cloud - create - lineno:623]: will not create cluster; already have cluster_id cl77t8sjiho7o5ih12ag
WARN [TBOT]      CLI parameters are overriding onboarding config from  config/config.go:471
INFO [TBOT]      Anonymous telemetry is not enabled. Find out more about Machine ID's anonymous telemetry at https://goteleport.com/docs/machine-id/reference/telemetry/ tbot/anonymous_telemetry.go:83
INFO [TBOT]      Successfully loaded bot identity, valid: after=2023-11-11T02:39:25Z, before=2023-11-11T08:40:24Z, duration=6h0m59s | kind=tls, renewable=false, disallow-reissue=false, roles=[bot-buildkite-robot], principals=[-teleport-internal-join], generation=0 tbot/tbot.go:372
WARN [TBOT]      Note: onboarding config ignored as identity was loaded from persistent storage tbot/tbot.go:379
INFO [TBOT]      Beginning renewal loop: ttl=6h0m0s interval=6h0m0s tbot/renew.go:721
INFO [TBOT]      Started watching for CA rotations tbot/ca_rotation.go:173
INFO [TBOT]      Attempting to generate new identity from token tbot/renew.go:505
INFO [AUTH]      Attempting registration via proxy server. auth/register.go:277
INFO [AUTH]      Attempting to register Bot with IAM method using regional STS endpoint auth/register.go:623
INFO [AUTH]      Successfully registered Bot with IAM method using regional STS endpoint auth/register.go:656
INFO [AUTH]      Successfully registered via proxy server. auth/register.go:284
INFO [TBOT]      Successfully renewed bot certificates, valid: after=2023-11-11T02:55:25Z, before=2023-11-11T08:56:24Z, duration=6h0m59s | kind=tls, renewable=false, disallow-reissue=false, roles=[bot-buildkite-robot], principals=[-teleport-internal-join], generation=0 tbot/renew.go:629
INFO [TBOT]      Successfully renewed impersonated certificates for directory /tmp/machine-id, valid: after=2023-11-11T02:55:26Z, before=2023-11-11T08:56:26Z, duration=6h1m0s | kind=tls, renewable=false, disallow-reissue=true, roles=[teleport-admin], principals=[root redpanda {{internal.logins}} -teleport-internal-join], generation=0 tbot/renew.go:692
INFO [TBOT]      Started watching for CA rotations tbot/ca_rotation.go:173
INFO [TBOT]      Persisted certificates successfully. One-shot mode enabled so exiting. tbot/renew.go:775
[INFO:2023-11-11 02:56:28,320]: RunnerClient: rptest.redpanda_cloud_tests.high_throughput_test.HighThroughputTest.test_max_connections: Running...
[WARNING - 2023-11-11 02:56:30,590 - high_throughput_test - test_max_connections - lineno:543]: Start swarming from 8 nodes: 3750 connections per node, 150 msg each producer, 29700 total connections expected
[WARNING - 2023-11-11 02:56:30,590 - high_throughput_test - test_max_connections - lineno:555]: Starting swarm node 1
[WARNING - 2023-11-11 02:56:31,730 - high_throughput_test - test_max_connections - lineno:555]: Starting swarm node 2
[WARNING - 2023-11-11 02:56:32,835 - high_throughput_test - test_max_connections - lineno:555]: Starting swarm node 3
[WARNING - 2023-11-11 02:56:33,934 - high_throughput_test - test_max_connections - lineno:555]: Starting swarm node 4
[WARNING - 2023-11-11 02:56:34,998 - high_throughput_test - test_max_connections - lineno:555]: Starting swarm node 5
[WARNING - 2023-11-11 02:56:36,112 - high_throughput_test - test_max_connections - lineno:555]: Starting swarm node 6
[WARNING - 2023-11-11 02:56:37,202 - high_throughput_test - test_max_connections - lineno:555]: Starting swarm node 7
[WARNING - 2023-11-11 02:56:38,191 - high_throughput_test - test_max_connections - lineno:555]: Starting swarm node 8
[WARNING - 2023-11-11 02:56:39,537 - high_throughput_test - test_max_connections - lineno:567]:   8.7s:   1073 connections (254/218/183/153/116/82/49/18) 
[WARNING - 2023-11-11 02:57:10,635 - high_throughput_test - test_max_connections - lineno:567]:  39.1s:   9212 connections (1258/1204/1187/1162/1121/1105/1125/1050) 
[WARNING - 2023-11-11 02:57:42,624 - high_throughput_test - test_max_connections - lineno:567]:  70.2s:  17773 connections (2326/2243/2258/2216/2171/2164/2258/2137) 
[WARNING - 2023-11-11 02:58:15,993 - high_throughput_test - test_max_connections - lineno:567]: 102.3s:  27215 connections (3477/3359/3466/3379/3364/3345/3502/3323) 
[WARNING - 2023-11-11 02:58:56,980 - high_throughput_test - test_max_connections - lineno:567]: 135.7s:  29964 connections (3744/3746/3745/3746/3747/3747/3745/3744) 
[WARNING - 2023-11-11 02:59:31,185 - high_throughput_test - test_max_connections - lineno:567]: 177.1s:  29966 connections (3746/3746/3745/3746/3747/3745/3747/3744) 
[WARNING - 2023-11-11 03:00:04,940 - high_throughput_test - test_max_connections - lineno:567]: 210.8s:  29973 connections (3746/3747/3746/3747/3749/3746/3747/3745) 
[WARNING - 2023-11-11 03:00:38,728 - high_throughput_test - test_max_connections - lineno:567]: 244.6s:  29975 connections (3746/3747/3746/3748/3749/3746/3747/3746) 
[WARNING - 2023-11-11 03:01:12,551 - high_throughput_test - test_max_connections - lineno:567]: 278.4s:  29705 connections (3689/3714/3704/3725/3732/3728/3690/3723) 
[WARNING - 2023-11-11 03:01:45,970 - high_throughput_test - test_max_connections - lineno:567]: 312.2s:  26777 connections (3147/3371/3201/3373/3458/3516/3184/3527) 
[WARNING - 2023-11-11 03:02:19,363 - high_throughput_test - test_max_connections - lineno:567]: 345.6s:  26380 connections (3096/3319/3160/3323/3403/3464/3119/3496) 
[WARNING - 2023-11-11 03:02:52,611 - high_throughput_test - test_max_connections - lineno:567]: 379.0s:  25884 connections (3045/3254/3102/3270/3331/3383/3054/3445) 
[WARNING - 2023-11-11 03:03:25,994 - high_throughput_test - test_max_connections - lineno:567]: 412.2s:  25591 connections (3018/3219/3069/3239/3293/3331/3000/3422) 
[WARNING - 2023-11-11 03:03:57,749 - high_throughput_test - test_max_connections - lineno:567]: 445.6s:  14382 connections (1704/1847/1643/1887/1784/1892/1618/2007) 
[WARNING - 2023-11-11 03:04:29,281 - high_throughput_test - test_max_connections - lineno:567]: 477.3s:  13090 connections (1552/1658/1501/1725/1605/1712/1473/1864) 
[WARNING - 2023-11-11 03:05:00,629 - high_throughput_test - test_max_connections - lineno:567]: 508.9s:  11352 connections (1322/1483/1286/1502/1368/1481/1278/1632) 
[WARNING - 2023-11-11 03:05:31,892 - high_throughput_test - test_max_connections - lineno:567]: 540.2s:  10569 connections (1222/1377/1196/1391/1289/1378/1176/1540) 
[WARNING - 2023-11-11 03:06:02,376 - high_throughput_test - test_max_connections - lineno:567]: 571.5s:   2553 connections (290/346/285/363/271/299/261/438) 
[WARNING - 2023-11-11 03:06:32,632 - high_throughput_test - test_max_connections - lineno:567]: 601.9s:      0 connections (0/0/0/0/0/0/0/0) 
[WARNING - 2023-11-11 03:06:32,731 - high_throughput_test - test_max_connections - lineno:584]: Waiting for messages
[WARNING - 2023-11-11 03:06:35,195 - high_throughput_test - test_max_connections - lineno:592]: 601.9s: Swarm nodes active: 0, total: 8, msg produced: 4498697 (100.05%), expected: 4496250
[WARNING - 2023-11-11 03:06:35,195 - high_throughput_test - test_max_connections - lineno:602]: Done swarming after 604.6048927307129s
[WARNING - 2023-11-11 03:06:35,195 - high_throughput_test - test_max_connections - lineno:605]: Stopping swarm
[WARNING - 2023-11-11 03:06:35,295 - high_throughput_test - test_max_connections - lineno:612]: Reached 29975 of 29700 needed
[WARNING - 2023-11-11 03:06:35,650 - high_throughput_test - test_max_connections - lineno:622]: Expected more than 4455000 (99.08256880733946%) messages out of 4496250, actual 4498697 (100.05442313038644%)
[INFO:2023-11-11 03:06:35,653]: RunnerClient: rptest.redpanda_cloud_tests.high_throughput_test.HighThroughputTest.test_max_connections: Tearing down...
[INFO:2023-11-11 03:06:42,193]: RunnerClient: rptest.redpanda_cloud_tests.high_throughput_test.HighThroughputTest.test_max_connections: PASS
[INFO:2023-11-11 03:06:42,194]: RunnerClient: rptest.redpanda_cloud_tests.high_throughput_test.HighThroughputTest.test_max_connections: Data: None
test_id:    rptest.redpanda_cloud_tests.high_throughput_test.HighThroughputTest.test_max_connections
status:     PASS
run time:   10 minutes 22.954 seconds
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
=============================================================================================================================================================================================================================================================================================================================
SESSION REPORT (ALL TESTS)
ducktape version: 0.8.18
session_id:       2023-11-11--012
run time:         10 minutes 24.606 seconds
tests run:        1
passed:           1
flaky:            0
failed:           0
ignored:          0
opassed:          0
ofailed:          0
=============================================================================================================================================================================================================================================================================================================================
```

vbotbuildovich commented:
<!--RETRIED FAILURES-->
ducktape was retried in https://buildkite.com/redpanda/redpanda/builds/40905#018bbcb3-fa90-4128-a62a-5c8e58d46c61
<!--3ab58b58-cd20-47a2-868b-a339975f46e7-->

andrewhsu commented on tests/rptest/redpanda_cloud_tests/high_throughput_test.py:
```diff
@@ -468,12 +468,26 @@ def _get_swarm_connections_count(self, nodes):
+        return _hwm
+
     @cluster(num_nodes=9)
     def test_max_connections(self):
```
nit: would be nice to have a pydoc comment that would describe the strategy of the test for validating max connections:
1. create client swarm connections up to the max advertised for the tier
2. count the messages sent and ensure at least a sizeable amount has gone through because just having a network connection is not solid enough to ensure data is flowing on the wire

andrewhsu reviewed (approved):
LGTM

i've validated this test passes in my dev env:
```console
ducktape   --debug   --globals=/home/ubuntu/redpanda/tests/globals.json   --cluster=ducktape.cluster.json.JsonCluster   --cluster-file=/home/ubuntu/redpanda/tests/cluster.json   --test-runner-timeout=36000000 tests/rptest/redpanda_cloud_tests/high_throughput_test.py::HighThroughputTest.test_max_connections
```

```
test_id:    rptest.redpanda_cloud_tests.high_throughput_test.HighThroughputTest.test_max_connections
status:     PASS
run time:   55 minutes 35.450 seconds
----------------------------
=====================
SESSION REPORT (ALL TESTS)
ducktape version: 0.8.18
session_id:       2023-11-13--004
run time:         55 minutes 35.468 seconds
tests run:        1
passed:           1
flaky:            0
failed:           0
ignored:          0
opassed:          0
ofailed:          0
=======================
```
